components/elements/Button.js

```diff
@@ -0,0 +1,53 @@
+/* eslint-disable react/prop-types */
+import styled, { css } from 'styled-components';
+
+const StyledButton = styled.button`
+  padding: 0.75em 1em;
+  font-weight: 400;
+  position: relative;
+  text-transform: uppercase;
+  cursor: pointer;
+  border: none;
+  border-radius: 50px;
+  font-size: 1.5rem;
+  transition: all ease-out 200ms;
+  letter-spacing: 1.5px;
+  &:disabled {
+    background-color: var(--Disabled);
+    pointer-events: none;
+    cursor: none;
+    color: var(--OnDisabled);
+    border: solid 2px var(--DisabledBorder);
+  }
+  ${({ styleProp }) =>
+    styleProp === 'primary' &&
+    css`
+      background-color: var(--Primary);
+      color: var(--OnMidground);
+    `};
```

Comment:
just adding to add something :D 
the background is either primary or transparent so you can also use sth like the below with correct punctuation
background-color: {({prop})} => prop === 'primary' ? 'var(--Primary)' : 'transparent' 

same for the border and padding as it only changes for one type

not sure if it saves you enough to be worth changing it up